Framework/Core/include/Framework/MessageContext.h

```diff
@@ -503,7 +503,7 @@ class MessageContext
             outputsPerChannel[mProxy.getOutputChannelIndex(message->route()).value].AddPart(std::move(part));
           }
         }
-        for (int ci = 0; ci < mProxy.getNumOutputChannels(); ++ci) {
+        for (int ci = 0; ci < (int)mProxy.getNumOutputChannels(); ++ci) {
```

Comment:
```suggestion
        for (size_t ci = 0; ci < mProxy.getNumOutputChannels(); ++ci) {
```

Reply:
I generally agree with you. I opted for a cast to avoid clashes with the `ChannelIndex` construction, so to limit the propagation of the change.

On the other hand I understand this is not code I developed, so I see two options here:
 1)  I follow the chain of changes to shut the warnings down to a correct fix that then you can check and approve
 1) I revert this fix and let owners taking care of warnings at their convenience

Both are fine with me, I see this is an annoying task, I wanted to contribute to the effort.